sci-libs/vtk/vtk-9.1.0.ebuild

```diff
@@ -0,0 +1,596 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+# TODO:
+# - add USE flag for remote modules? Those modules can be downloaded
+#	properly before building.
+# - replace usex by usev once we bump to EAPI 8
+
+PYTHON_COMPAT=( python3_{8..10} )
+WEBAPP_OPTIONAL=yes
+WEBAPP_MANUAL_SLOT=yes
+
+inherit check-reqs cmake cuda java-pkg-opt-2 python-single-r1 toolchain-funcs virtualx webapp
+
+# Short package version
+MY_PV="$(ver_cut 1-2)"
+
+DESCRIPTION="The Visualization Toolkit"
+HOMEPAGE="https://www.vtk.org/"
+SRC_URI="
+	https://www.vtk.org/files/release/${MY_PV}/VTK-${PV}.tar.gz
+	https://www.vtk.org/files/release/${MY_PV}/VTKData-${PV}.tar.gz
+	https://www.vtk.org/files/release/${MY_PV}/VTKDataFiles-${PV}.tar.gz
+	doc? ( https://www.vtk.org/files/release/${MY_PV}/vtkDocHtml-${PV}.tar.gz )
+	examples? (
+		https://www.vtk.org/files/release/${MY_PV}/VTKLargeData-${PV}.tar.gz
+		https://www.vtk.org/files/release/${MY_PV}/VTKLargeDataFiles-${PV}.tar.gz
+	)
+	test? (
+		https://www.vtk.org/files/release/${MY_PV}/VTKLargeData-${PV}.tar.gz
+		https://www.vtk.org/files/release/${MY_PV}/VTKLargeDataFiles-${PV}.tar.gz
+	)
+"
+S="${WORKDIR}/VTK-${PV}"
+
+LICENSE="BSD LGPL-2"
+SLOT="0/${MY_PV}"
+KEYWORDS="~amd64 ~arm ~arm64 ~x86 ~amd64-linux ~x86-linux"
+# TODO: Like to simplifiy these. Mostly the flags related to Groups, plus
+# maybe some flags related to Kits and a few other needed flags.
+IUSE="all-modules +boost cuda debug doc examples +ffmpeg +gdal imaging java
+	mpi mysql odbc openmp postgres python qt5 +rendering tbb test +threads
+	tk video_cards_nvidia views web"
+
+RESTRICT="!test? ( test )"
+
+REQUIRED_USE="
+	all-modules? ( boost ffmpeg gdal imaging mysql odbc postgres qt5 rendering views )
+	cuda? ( video_cards_nvidia )
+	java? ( rendering )
+	python? ( ${PYTHON_REQUIRED_USE} )
+	qt5? ( rendering )
+	tk? ( rendering python )
+	web? ( python )
+"
+
+RDEPEND="
+	app-arch/lz4:=
+	app-arch/xz-utils
+	dev-db/sqlite:3
+	dev-libs/double-conversion:=
+	dev-libs/expat
+	dev-libs/icu:=
+	dev-libs/jsoncpp:=
+	dev-libs/libxml2:2
+	dev-libs/pugixml
+	media-libs/freetype
+	media-libs/libogg
+	media-libs/libpng:=
+	media-libs/libtheora
+	media-libs/tiff
+	<sci-libs/hdf5-1.12:=[mpi=]
```

Comment:
I'm trying to install vtk-9.0.3 and I see that hdf5 needs to be installed with USE=hl for Vtk:
```
-- HDF5: Using hdf5 compiler wrapper to determine C configuration
-- Could NOT find HDF5 (missing: HDF5_HL_LIBRARIES) (found version "1.10.5")
CMake Error at CMake/vtkModule.cmake:4145 (message):
  Could not find the HDF5 external dependency.
Call Stack (most recent call first):
  CMake/vtkModule.cmake:4695 (vtk_module_find_package)
  CMake/vtkModule.cmake:4569 (vtk_module_third_party_external)
  ThirdParty/hdf5/CMakeLists.txt:8 (vtk_module_third_party)
```
It would also be nice if the restriction on the version could be lifted as well.

Reply:
BTW, I can help with testing CUDA support, as I have a recent Nvidia card.

Reply:
Thanks for the hint. That's correct, see https://gitlab.kitware.com/vtk/vtk/-/blob/v9.0.3/ThirdParty/hdf5/CMakeLists.txt#L17. I've missed this so far, probably because it's set by default.

As for hdf5-1.12.5, this should be supported by 9.1.0, but I couldn't get it to work. Any insights into this, are welcome. I wouldn't bother with it for now, as there were more issue concerning hdf5-1.12 when I started working on 9.1.0. It will not work, however, for 9.0.3.